File: js/day1/main.js
const fs = require('fs');

function sumDigits(s) {
    let numz = '';
    let tempArr = [];

    for (let char of s) {
        if (/\d/.test(char)) {
            const digit = parseInt(char);

            if (!isNaN(digit)) {
                tempArr.push(digit);
                numz = tempArr[0] + '' + tempArr[tempArr.length - 1];
            }
        }
    }

    return numz;
}

function main() {
    try {
        const data = fs.readFileSync('../../go/day1/lefile1.txt', 'utf8').split('\n');
        let total = 0;

        for (let line of data) {
            const lineSum = sumDigits(line);

            console.log(`Line: ${line}, Double Number: ${lineSum}`);
            const lineDigit = parseInt(lineSum);

            if (!isNaN(lineDigit)) {
                total += lineDigit;
            }
        }

        console.log(`Total: ${total}`);
    } catch (err) {
        console.error('Error reading file:', err);
    }
}

main();

File: go/day1/main.go
package sumDigits

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"unicode"
)

func sumDigits(s string) string {
	var numz string
	var tempArr []int

	for _, char := range s {
		if unicode.IsDigit(char) {
			digit, err := strconv.Atoi(string(char))

			if err != nil {
				fmt.Println(err)
			} else {
				tempArr = append(tempArr, digit)
				numz = strconv.Itoa(tempArr[0]) + strconv.Itoa(tempArr[len(tempArr)-1])
			}
		}
	}
	return numz
}

func main() {
	file, err := os.Open("./lefile1.txt")
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	total := 0

	for scanner.Scan() {
		line := scanner.Text()

		lineSum := sumDigits(line)

		fmt.Printf("Line: %s, Double Number: %s\n", line, lineSum)
		linedigit, err := strconv.Atoi(string(lineSum))

		if err != nil {
			fmt.Println(err)
		} else {
			total += linedigit
		}

	}
	fmt.Printf("Total: %s\n", total)

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading file:", err)
	}
}
